### PR TITLE
Add dockerfile and instructions

### PR DESCRIPTION
**Pull request by d4v3y0rk**
This adds a working dockerfile to the project and instructions on how to use it in the readme. 

(This could also be used to generate a docker image that can be pushed to Dockerhub for people to use if the project team so desires)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,16 @@ Cryptographic operations in CyberChef should not be relied upon to provide secur
 
 [A live demo can be found here][1] - have fun!
 
+## Running in Docker
```
Comment (GCHQ77703) [inline README.md:25]: This section might be better put lower down in the page.  People will likely want to read what the project does before learning how to run it.  Maybe around the contributing section?

Comment (d4v3y0rk) [reply README.md:25]: My thought process here was that running it locally goes hand in hand with the live demo but I can put it where ever you want it to go.

Comment (d4v3y0rk) [reply README.md]: I moved the section to where it was suggested.

**Comment by d4v3y0rk** [comment]
@GCHQ77703 Just pinging to make sure this doesn't get forgotten about or fall too far down the list.

**Comment by oscrx** [comment]
I would suggest adding a [staged build](https://docs.docker.com/develop/develop-images/multistage-build/) for a leaner container.

Also building with [dockerx](https://docs.docker.com/buildx/working-with-buildx/) and releasing for multiple architectures (amd64, arm64, etc.)
Would be a nice feature, required for Raspberry pi's, M1 Apple devices (Docker for mac doesn't support Rosetta 2 yet), etc.
[An example within a Github actions pipeline](https://github.com/oscrx/caddy/blob/54e9555b00e1a3cfb3c0c834b0441ede9f58319f/.github/workflows/docker.yml)

Publishing to [Github container repository](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) (ghcr.io) is very easy and can be done with only the [Github token](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret) already available in the Github actions pipeline. (see example above)

For docker hub; [credentials](https://docs.docker.com/docker-hub/access-tokens/) need to be manually added by a project admin.

I might do a PR on [your](https://github.com/d4v3y0rk/CyberChef/tree/add-dockerfile) repo if I have the time tonight, if that is even possible.

**Comment by d4v3y0rk** [comment]
> I would suggest adding a [staged build](https://docs.docker.com/develop/develop-images/multistage-build/) for a leaner container.
> 
> Also building with [dockerx](https://docs.docker.com/buildx/working-with-buildx/) and releasing for multiple architectures (amd64, arm64, etc.)
> Would be a nice feature, required for Raspberry pi's, M1 Apple devices (Docker for mac doesn't support Rosetta 2 yet), etc.
> [An example within a Github actions pipeline](https://github.com/oscrx/caddy/blob/54e9555b00e1a3cfb3c0c834b0441ede9f58319f/.github/workflows/docker.yml)
> 
> Publishing to [Github container repository](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) (ghcr.io) is very easy and can be done with only the [Github token](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret) already available in the Github actions pipeline. (see example above)
> 
> For docker hub; [credentials](https://docs.docker.com/docker-hub/access-tokens/) need to be manually added by a project admin.
> 
> I might do a PR on [your](https://github.com/d4v3y0rk/CyberChef/tree/add-dockerfile) repo if I have the time tonight, if that is even possible.

all that would be great to add. if they would come back around and merge this we could add on some of those features later. Iterate.

**Comment by nileshtrivedi** [comment]
This would be really helpful for easier deployment in container environments (like CapRover for self-hosting).

**Comment by d4v3y0rk** [comment]
🤷🏼‍♂️ I don't know why they won’t merge it. 

Dave York
412-944-9675
***@***.***

**Comment by lnodaba** [comment]
This doesn't build. I get this error:
image

CC: @d4v3y0rk

**Comment by a3957273** [comment]
Supplanted by #1699

**Comment by a3957273** [comment]
Docker images are now automatically built and provided here: https://github.com/gchq/CyberChef/pkgs/container/cyberchef

**Pull request closed without merging**